src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerCommitImage.groovy

```diff
@@ -67,8 +67,8 @@ class DockerCommitImage extends DockerExistingContainer {
 
     @Override
     void runRemoteCommand(dockerClient) {
-        logger.quiet "Commiting image for container '${getContainerId()}'."
-        def commitCmd = dockerClient.commitCmd(getContainerId())
+        logger.quiet "Commiting image for container '${getContainerId().get()}'."
+        def commitCmd = dockerClient.commitCmd(getContainerId().get())
```

Comment:
Can we go with what I did for #720 here? I think it would be a good idea to start a new standard in that we need to check that the containerId was set and fail accordingly.

Reply:
The `get()` method will actually throw an exception if no value was set.

Reply:
`getOrNull()`?

Reply:
I only say to use that as it _may_ not be exactly obvious when buried within the gradle stacktrace at least in this case we can control the message.

Reply:
I updated the code with the implementation from #720

Reply:
@cdancy The method `get()` will throw an exception if no value was set. The method `getOrNull()` will not throw an exception but will return null if no value is net. I'd try to create a test and see how the error message looks like. Otherwise, we are likely just duplicating code.

Reply:
I added a couple of functional tests for the cases where the container is missing, or is non existing. 

In the case of the missing container, the test does not show case the `getOrNull()` behaviour because the task validation kicks in first, and since `DockerExistingContainer#containerId` input is not optional, the validation fails the build with error `No value has been specified for property 'containerId'`.

In the case of the non existing container, then we get the error message from the docker daemon.